Intermittent failure with fs mount in TestFileStorageMountUser_SmokeTest

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://Rook-io.slack.com).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
On March 1st, we started seeing intermittent failures in the master builds in the test `TestFileStorageMountUser_SmokeTest`. 

The issue has been seen in builds 977, 978, 979, and [984](https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/master/984/pipeline/41/) thus far. Builds 980-983 did not hit this issue. Perhaps coincidentally, each failure always happened on K8s 1.11, but in the last build it also hit on K8s 1.12.

```
--- FAIL: TestSmokeSuite (462.26s)
    --- PASS: TestSmokeSuite/TestBlockCSI_SmokeTest (0.01s)
    --- PASS: TestSmokeSuite/TestBlockStorage_SmokeTest (116.38s)
    --- FAIL: TestSmokeSuite/TestFileStorageMountUser_SmokeTest (16.87s)
        require.go:765: 
            	Error Trace:	filesystem_mountuser_test.go:65
            	            				filesystem_mountuser_test.go:46
            	            				smoke_test.go:94
            	Error:      	Expected nil, but got: &errors.errorString{s:"kubectl exec command bash failed on pod rook-ceph-tools in namespace smoke-ns: Failed to run: kubectl [exec -n smoke-ns rook-ceph-tools -- bash -c mount -t ceph -o mds_namespace=smoke-test-fs-mountuser,name=admin,secret=$(grep key /etc/ceph/keyring | awk '{print $3}') $(grep mon_host /etc/ceph/ceph.conf | awk '{print $3}'):/ /tmp/testrook] : Failed to complete 'kubectl': exit status 2. . output: mount error 2 = No such file or directory\ncommand terminated with exit code 2"}
            	Test:       	TestSmokeSuite/TestFileStorageMountUser_SmokeTest
```

@BlaineEXE Could you take a look? The last PR merged around this time was #2702, though it's not clearly related to that either.

**Expected behavior:**
No intermittent CI issues

--- a/tests/framework/utils/k8s_helper.go
+++ b/tests/framework/utils/k8s_helper.go
@@ -282,15 +282,25 @@ type kuserContext struct {
 }
 
 func (k8sh *K8sHelper) Exec(namespace, podName, command string, commandArgs []string) (string, error) {
-	args := []string{"exec", "-n", namespace, podName, "--", command}
-	args = append(args, commandArgs...)
-	result, err := k8sh.Kubectl(args...)
-	if err != nil {
-		return "", fmt.Errorf("kubectl exec command %s failed on pod %s in namespace %s: %+v. output: %s",
-			command, podName, namespace, err, result)
-	}
+	return k8sh.ExecWithRetry(1, namespace, podName, command, commandArgs)
+}
 
-	return result, nil
+// ExecWithRetry will attempt to run a command "retries" times, waiting 3s between each call. Upon success, returns the output.
+func (k8sh *K8sHelper) ExecWithRetry(retries int, namespace, podName, command string, commandArgs []string) (string, error) {
+	var err error
+	for i := 0; i < retries; i++ {
+		args := []string{"exec", "-n", namespace, podName, "--", command}
+		args = append(args, commandArgs...)
+		var result string
+		result, err = k8sh.Kubectl(args...)
+		if err == nil {
+			return result, nil
+		}
+		if i < retries-1 {
+			time.Sleep(3 * time.Second)
+		}
+	}
+	return "", fmt.Errorf("kubectl exec command %s failed on pod %s in namespace %s. %+v", command, podName, namespace, err)
 }
 
 // ResourceOperationFromTemplate performs a kubectl action from a template file after replacing its context

--- a/tests/integration/filesystem_mountuser_test.go
+++ b/tests/integration/filesystem_mountuser_test.go
@@ -61,7 +61,7 @@ func createFilesystemMountCephCredentials(helper *clients.TestClient, k8sh *util
 	logger.Info("Creating /foo directory on CephFS")
 	_, err = k8sh.Exec(namespace, "rook-ceph-tools", "mkdir", []string{"-p", utils.TestMountPath})
 	require.Nil(s.T(), err)
-	_, err = k8sh.Exec(namespace, "rook-ceph-tools", "bash", []string{"-c", fmt.Sprintf("mount -t ceph -o mds_namespace=%s,name=admin,secret=$(grep key /etc/ceph/keyring | awk '{print $3}') $(grep mon_host /etc/ceph/ceph.conf | awk '{print $3}'):/ %s", filesystemName, utils.TestMountPath)})
+	_, err = k8sh.ExecWithRetry(3, namespace, "rook-ceph-tools", "bash", []string{"-c", fmt.Sprintf("mount -t ceph -o mds_namespace=%s,name=admin,secret=$(grep key /etc/ceph/keyring | awk '{print $3}') $(grep mon_host /etc/ceph/ceph.conf | awk '{print $3}'):/ %s", filesystemName, utils.TestMountPath)})
 	require.Nil(s.T(), err)
 	_, err = k8sh.Exec(namespace, "rook-ceph-tools", "mkdir", []string{"-p", fmt.Sprintf("%s/foo", utils.TestMountPath)})
 	require.Nil(s.T(), err)

